i3pystatus/cpu_usage.py

```diff
@@ -29,20 +32,39 @@ class CpuUsage(IntervalModule):
     exclude_average = False
     interval = 1
     color = None
+    dynamic_color = False
+    upper_limit = 100
     settings = (
         ("format", "format string."),
         ("format_all", ("format string used for {usage_all} per core. "
                         "Available formaters are {core} and {usage}. ")),
         ("exclude_average", ("If True usage average of all cores will "
                              "not be in format_all.")),
-        ("color", "HTML color code #RRGGBB")
+        ("color", "HTML color code #RRGGBB"),
+        ("dynamic_color", "Set color dynamically based on CPU usage. Note: this overrides color_up"),
+        ("start_color", "Hex or English name for start of color range, eg '#00FF00' or 'green'"),
+        ("end_color", "Hex or English name for end of color range, eg '#FF0000' or 'red'")
     )
 
     def init(self):
         self.prev_total = defaultdict(int)
         self.prev_busy = defaultdict(int)
         self.formatter = Formatter()
 
+        self.key = re.findall('usage_cpu\d+', self.format)
+        if len(self.key) == 1:
+            self.key = self.key[0]
+        else:
+            self.key = 'usage_cpu'
+
+        if not self.color:
```

Comment:
No need to set this explicitly. It's probably best to just not set the color key in the output if no color is specified and dynamic_color is False.